plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -687,6 +687,7 @@ class producer_plugin_impl : public std::enable_shared_from_this<producer_plugin
    std::map<chain::public_key_type, signature_provider_type> _signature_providers;
    chain::bls_pub_priv_key_map_t                     _finalizer_keys; // public, private
    std::set<chain::account_name>                     _producers;
+   named_thread_pool<struct prod>                    _prod_thread_pool; // for timers
```

Comment:
Isn't this premature before understanding the design of #929[^1]? imo this just adds unnecessary timer accuracy error -- as long we have an io_service running on the main thread, main thread timers should expire there. That seems preferable even if we don't expose the io_service (we could hide the io_service from being used for anything other than timers via a `appbase::make_timer()` sort of pattern). Otherwise we've needlessly made these timers require waking 2 threads instead of 1.

[^1]: strictly speaking not even #929 since refactoring the priority queue and its executor doesn't imply removing the main thread io_context; that could be its own issue

Reply:
I'll revert this and go with a `make_timer` approach.